cristin-project/src/main/java/no/unit/nva/cristin/projects/query/organization/QueryCristinOrganizationProjectHandler.java

```diff
@@ -93,26 +80,15 @@ protected QueryCristinOrganizationProjectHandler(QueryCristinOrganizationProject
     @Override
     protected SearchResponse<NvaProject> processInput(Void input, RequestInfo requestInfo, Context context)
             throws ApiGatewayException {
-
-        validateHasIdentifierPathParameter(requestInfo);
-        validateQueryParameterKeys(requestInfo);
-
-        var requestQueryParameters = extractQueryParameters(requestInfo);
-
-        return cristinApiClient.listOrganizationProjects(requestQueryParameters);
-    }
-
-    private Map<String, String> extractQueryParameters(RequestInfo requestInfo)
-            throws BadRequestException {
-        var requestQueryParameters = new ConcurrentHashMap<>(Map.of(
-                CRISTIN_QUERY_PARAMETER_PARENT_UNIT_ID, getValidId(requestInfo),
-                LANGUAGE, getValidLanguage(requestInfo),
-                PAGE, getValidPage(requestInfo),
-                NUMBER_OF_RESULTS, getValidNumberOfResults(requestInfo)));
-        //From here
-        VALID_QUERY_PARAM_NO_VALIDATION.forEach(paramName -> putOrNotQueryParameterOrEmpty(requestInfo,
-                paramName, requestQueryParameters));
-        return requestQueryParameters;
+        var cristinQuery =
+            CristinQuery.builder()
+                .fromRequestInfo(requestInfo)
+                .withRequiredParameters(PATH_ORGANISATION, PATH_PROJECT, LANGUAGE, PAGE_CURRENT, PAGE_ITEMS_PER_PAGE)
```

Comment:
required or default values if missing?

Reply:
actually both, if a value is required and has default values, it will be set, if it isn't required, it will not get default values.
(when creating URLs you don't usually want to add default values, so I had to differentiate when to add default values or not)